components/sidebar_right/sidebar_right.jsx

```diff
@@ -154,92 +152,47 @@ export default class SidebarRight extends React.PureComponent {
-        }
-
-        let channelDisplayName = '';
-        if (rhsChannel) {
-            channelDisplayName = rhsChannel.display_name;
-        }
-
-        if (searchVisible) {
-            content = (
-                <div className='sidebar--right__content'>
-                    <div className='search-bar__container channel-header alt'>{searchForm}</div>
-                    <SearchResults
-                        isMentionSearch={isMentionSearch}
-                        isFlaggedPosts={isFlaggedPosts}
-                        isPinnedPosts={isPinnedPosts}
-                        toggleSize={this.toggleSize}
-                        shrink={this.onShrink}
-                        channelDisplayName={channelDisplayName}
-                        isOpened={this.state.isOpened}
-                        updateSearchTerms={this.handleUpdateSearchTerms}
-                        isSideBarExpanded={isExpanded}
-                    />
-                </div>
-            );
-        } else if (postRightVisible) {
+        switch (true) {
+        case postRightVisible:
+            isSbrExpanded = isExpanded;
             content = (
-                <div className='post-right__container'>
-                    <FileUploadOverlay overlayType='right'/>
-                    <div className='search-bar__container channel-header alt'>{searchForm}</div>
-                    <RhsThread
-                        previousRhsState={previousRhsState}
-                        currentUserId={currentUserId}
-                        toggleSize={this.toggleSize}
-                        shrink={this.onShrink}
-                    />
-                </div>
-            );
-        } else if (isPluginView) {
-            content = (
-                <div className='post-right__container'>
-                    <div className='search-bar__container channel-header alt'>{searchForm}</div>
-                    <RhsPlugin/>
-                </div>
-            );
-        } else if (postCardVisible) {
-            content = (
-                <div className='post-right__container'>
-                    <div className='search-bar__container channel-header alt'>{searchForm}</div>
-                    <RhsCard previousRhsState={previousRhsState}/>
-                </div>
+                <RhsThread
+                    previousRhsState={previousRhsState}
+                    currentUserId={currentUserId}
+                    toggleSize={this.toggleSize}
+                    shrink={this.onShrink}
+                />
             );
-        }
-
-        if (!content) {
-            expandedClass = '';
+            break;
+        case postCardVisible:
+            isSbrExpanded = isExpanded;
+            content = <RhsCard previousRhsState={previousRhsState}/>;
+            break;
+        case isPluginView:
+            isSbrExpanded = isExpanded;
+            content = <RhsPlugin/>;
+            break;
+        case searchVisible:
+            isSbrExpanded = isExpanded;
```

Comment:
Note: switch(true) is nice, but I think it will be more clear for the `isSbrExpanded` to be assigned more explicitly, so I would extract it.

Reply:
done!